src/nvim/os_unix_defs.h

```diff
@@ -1,216 +0,0 @@
-#ifndef NVIM_OS_UNIX_DEFS_LEGACY_H
-#define NVIM_OS_UNIX_DEFS_LEGACY_H
-
-/*
- * VIM - Vi IMproved	by Bram Moolenaar
- *
- * Do ":help uganda"  in Vim to read copying and usage conditions.
- * Do ":help credits" in Vim to see a list of people who contributed.
- */
-
-#include <stdio.h>
-#include <ctype.h>
-
-#include <sys/types.h>
-#include <sys/stat.h>
-
-#include <stdlib.h>
-
-#ifdef HAVE_UNISTD_H
-# include <unistd.h>
-#endif
-
-#ifdef HAVE_SYS_PARAM_H
-# include <sys/param.h>     /* defines BSD, if it's a BSD system */
-#endif
-
-/* The number of arguments to a signal handler is configured here. */
-/* It used to be a long list of almost all systems. Any system that doesn't
- * have an argument??? */
-#define SIGHASARG
-
-/* List 3 arg systems here. I guess __sgi, please test and correct me. jw. */
-
-#ifdef SIGHASARG
-# ifdef SIGHAS3ARGS
-#  define SIGDEFARG(s)  (int s, int sig2, struct sigcontext *scont)
-#  define SIGDUMMYARG   0, 0, (struct sigcontext *)0
-# else
-#  define SIGDEFARG(s)  (int s)
-#  define SIGDUMMYARG   0
-# endif
-#else
-# define SIGDEFARG(s)  (void)
-# define SIGDUMMYARG
-#endif
-
-#if !defined(HAVE_SYS_TIME_H) || defined(TIME_WITH_SYS_TIME)
-# include <time.h>          /* on some systems time.h should not be
-                               included together with sys/time.h */
-#endif
-#ifdef HAVE_SYS_TIME_H
-# include <sys/time.h>
-#endif
-
-#include <signal.h>
-
-#if defined(DIRSIZ) && !defined(MAXNAMLEN)
-# define MAXNAMLEN DIRSIZ
-#endif
-
-#if defined(UFS_MAXNAMLEN) && !defined(MAXNAMLEN)
-# define MAXNAMLEN UFS_MAXNAMLEN    /* for dynix/ptx */
-#endif
-
-#if defined(NAME_MAX) && !defined(MAXNAMLEN)
-# define MAXNAMLEN NAME_MAX         /* for Linux before .99p3 */
```

Comment:
The comment here (which you removed...) [suggests](https://lkml.org/lkml/2000/3/25/32) that this define exists to support 15-year-old linux, which I don't think we support. 

So, again, moving things around requires doing this research and providing reasons for your changes.